MAP/src/App.js

```diff
@@ -47,243 +50,289 @@ const getDaysDiff = (date1, date2) => {
+        if (playerState === PLAYING) {
+          loop = setInterval(() => {
+            if (playerState === PAUSED || formattedSelectedDate === endDate) {
+              console.log('Stopped');
+              clearInterval(loop);
+            } else {
+              console.log('Still looping');
+              setSelectedDate(format(
+                addDays(formattedSelectedDate, 1),
+                'yyyy-MM-dd',
+              ))
+            }
+          }, playSpeed);
+        }
+    
+        return () => clearInterval(loop);
+    },
+    [selectedDate,endDate,playerState],
+  ) 
 
-  setEndDate = (endDate) => {
-    this.setState({
-      ...this.state,
-      endDate,
-    });
-  };
+  const updateIsDark = useCallback(
+    () => {
+        const darkModePreference = window.localStorage.getItem('darkmode');
+    
+        if (!darkModePreference) {
+          const isDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
+          setIsDark(isDarkTheme.toString());
+          document.getElementsByTagName('html')[0].classList.add('dark');
+          window.localStorage.setItem('darkmode', 'true');
+        }
+        if (darkModePreference === 'true') {
+          document.getElementsByTagName('html')[0].classList.add('dark');
+          setIsDark("true");
+        } else if (darkModePreference === 'false') {
+          setIsDark("false");
+        }
+    },
+    [],
+  ) 
 
-  pausePlayerState = () => {
-    const { endDate, selectedDate, startDate } = this.state;
-    const formattedSelectedDate = new Date(selectedDate);
-    if (
-      formattedSelectedDate.getDate() === endDate.getDate() &&
-      formattedSelectedDate.getMonth() === endDate.getMonth() &&
-      formattedSelectedDate.getFullYear() === endDate.getFullYear()
-    ) {
-      alert('Ended');
-      this.setState({
-        ...this.state,
-        playerStates: PAUSED,
-        selectedDate: format(startDate, 'yyyy-MM-dd'),
-      });
-    }
-  };
 
-  toggleState = (newState) => {
-    this.setState({
-      ...this.state,
-      playerState: newState,
-    });
-  };
+  const closeDialog = useCallback(
+    () => {
+        setDialog(prevState => ({
+          ...prevState,
+          opened: false, template: '', title: '' 
+          }));
+    },
+    [],
+  ) 
 
-  setPlayerState = () => {
-    const { selectedDate, endDate, playerState } = this.state;
-    const formattedSelectedDate = new Date(selectedDate);
-    let loop = null;
-    if (playerState === PLAYING) {
-      loop = setInterval(() => {
-        if (playerState === PAUSED || formattedSelectedDate === endDate) {
-          console.log('Stopped');
-          clearInterval(loop);
-        } else {
-          console.log('Still looping');
-          this.setState({
-            ...this.state,
-            selectedDate: format(
-              addDays(formattedSelectedDate, 1),
-              'yyyy-MM-dd',
-            ),
-          });
+  const openDialog = useCallback(
+    (props) => {
```

Comment:
`'props' is already declared in the upper scope`
https://sonarcloud.io/project/issues?id=TheIOFoundation_ProjectLockdown&open=AXmgINXzDWFvZvlR_NPD&pullRequest=380&resolved=false&types=CODE_SMELL

Reply:
okay I will address